docs/source/install/all_in_one.rst

```diff
@@ -0,0 +1,136 @@
+All-in-one Installation
+=======================
+|st2| provides an all-in-one installer aimed at assisting users with the initial setup and configuration. The installer comes pre-bundled in a number of different provisioning options for convenience, or can also be manually deployed and installed manually on a server.
+
+Pre-Requsites
+-------------
+Before getting started, it is necessary to do a bit of pre-planning to integrate |st2| into an environment. Below is a checklist of items to prepare before running the installer.
+
+#. A server. See below for the list of pre-bundled options, or bring your own.
+#. Hostname of new server setup with DNS.
+#. Admin Password.
+#. Administrative SSH Public/Private Keys *(Optional)*
+#. SSL Certificates *(Optional)*
+#. Chat service connection credentials *(Optional)*
+
+The initial setup is designed to configure itself to a freshly deployed server with no additional services running. If you do not have or do not wish to provide SSL and SSH keys during the initial setup, |st2| will generate random new keys automatically.
+
+Sizing the Server
+-----------------
+A standard all-in-one installation has all of the various components of |st2|, as well as the supporting infrastructure components. While the system can operate with less equipped servers, these are recommended for the best experience while testing or deploying |st2|.
+
+Testing
+~~~~~~~
+* Single or Dual CPU system
+* At least 4GB of RAM
```

Comment:
Well, I'm glad this spurred the conversation I hoped it would.

The fact of the matter is that there are two different databases running (Mongo+MySQL), as well as all of the various processes. I actually started at the low bar of 2GB, but thought I would put out the 4GB to see what everyone said.

At the very least, I don't want folks starting up with >1GB, because we have seen folks try to start up in Docker containers and small ec2 instances, only to be surprised that things didn't work.

I think 2GB is reasonable, but does that create the experience we hope? Right now, we only have feelings and anecdotes here. We're coming up on 1.0, and we need to have hard data before then. I know we don't now, but that cannot be for much longer. /cc @dzimine